### ISSUE TITLE
Weird error message on case-insensitive file systems

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#4671]
Weird error on case-insensitive file systems (like Mac OS X):
```agda
open import Agda.PRIMITIVE
```
```
The file
.../Agda/Primitive.agda
can be accessed via several project roots. Both Agda.Primitive and
Agda.PRIMITIVE point to this file.
when scope checking the declaration
  open import Agda.PRIMITIVE
```

**Comment by andreasabel** [comment]
It seems it is sufficient to fix the error message.
https://github.com/agda/agda/blob/80d910b4c6da41818e6ffa9c750dcd40a5345520/src/full/Agda/TypeChecking/Errors.hs#L703-L708
When both top level module names are equal modulo case, the error should say:
```
Case mismatch when referring to file ../Agda/Primitive.agda with module name Agda.PRIMITIVE
```
A well-maintained library for comparing modulo case seems https://hackage.haskell.org/package/case-insensitive .  (Even part of the Haskell platform.)

**Comment by andreasabel** [comment]
Something broke (not adding an OS-dependent test case fired back on me...).
Now I get:
```
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE_VERBOSE__, called at src/full/Agda/TypeChecking/Monad/Signature.hs:731:16 in Agda-2.6.2-IymqxoxZr9b2Sz3kgFx3aF:Agda.TypeChecking.Monad.Signature
```
The full error is:
```
$ agda -v impossible:100  Issue4671.agda 
Checking Issue4671 (/Users/abel/tmp/agda/Issue4671.agda).
 Checking Agda.PRIMITIVE (/Users/abel/.cabal/share/x86_64-osx-ghc-8.10.2/Agda-2.6.2/lib/prim/Agda/PRIMITIVE.agda).
Ambiguous name: Agda.Primitive.Set
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE_VERBOSE__, called at src/full/Agda/TypeChecking/Monad/Signature.hs:731:16 in Agda-2.6.2-IymqxoxZr9b2Sz3kgFx3aF:Agda.TypeChecking.Monad.Signature
```

**Comment by andreasabel** [comment]
Not too suprisingly, I broke this myself by messing with the path canonicalization:

be5cafa4a47d5952d3bc259ba076ce042c67027b is the first bad commit
Date:   Tue Aug 11 13:59:07 2020 +0200

[ fixed #4828 ] by deferring canonicalizePath
    
    Rather than eagerly resolving symlinks, which leads to wrong placement
    of .agdai files, we only resolve them when testing whether two filenames
    point to the same file, and when serializing AbsolutePath.
